jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/SnackbarTests.java

```diff
@@ -0,0 +1,65 @@
+package io.github.epam.angular.tests.elements.common;
+
+import io.github.epam.TestsInit;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static io.github.com.StaticSite.angularPage;
+import static io.github.com.pages.AngularPage.snackbarSection;
+import static io.github.epam.angular.tests.elements.BaseValidationsUtils.duration;
+import static io.github.epam.site.steps.States.shouldBeLoggedIn;
+
+public class SnackbarTests extends TestsInit {
+
+    private static final String MESSAGE = "Test Message";
+    private static final String ACTION = "Test Action";
+    private static final int DURATION = 5;
+
+    @BeforeMethod
+    public void before() {
+        shouldBeLoggedIn();
+        angularPage.shouldBeOpened();
+    }
+
+    @Test
+    public void checkBasicSnackbarTest() {
+        snackbarSection.messageInput.setValue(MESSAGE);
```

Comment:
In almost all test cases we interact with SnackbarSection elements such as input, button etc. To my mind all methods that we use in tests should be Snackbar methods. For example this test should look like this:
public void checkBasicSnackbarTest() {
basicSnackbar.setMessage(MESSAGE);
basicSnackbar.setAction(ACTION);
basicSnackbar.showSnack(); // click on button here
basicSnackbar.has().snackMessage(MESSAGE);
basicSnackbar.has().snackAction(ACTION);
}

Reply:
As far as I understand these elements (two text fields and button) are used to provide parameters to Snackbar and open it. They are not part of Snackbar and therefore located outside its implementation.

![image](https://user-images.githubusercontent.com/68592812/88514438-c4dbdd00-cff2-11ea-9874-f6a75b29fcae.png)

Reply:
Yes, I know. I have the same behavior in the MaterialSelector implementation. Also there is another approach see: AutoComplete element.